### PR TITLE
use go workspace

### PR DESCRIPTION
**Pull request by codeboten**
This change removes the need for `replace` statements in all modules within this repository. Some additional changes needed to make this work:
    
- ~update example docker base image~
- ~update compliance tests to avoid issue w/ go workspace~

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@codeboten are we going to have this 👯 ?

**Comment by codeboten** [comment]
@bogdandrutu yes I'll tackle this as soon as I can get https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12918 merged

**Review by jmacd — APPROVED** [review]

File: examples/tracing/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17 AS build
+FROM golang:1.18 AS build
```
Comment (jpkrohling) [inline examples/tracing/Dockerfile:1]: Should we just go for 1.19 instead?

File: go.work [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+	./receiver/zipkinreceiver
+	./receiver/zookeeperreceiver
+	./testbed
+	./testbed/mockdatareceivers/mockawsxrayreceiver
+)
+
+// 	references to modules that are in this repository with their relative paths
+// so that we always build with current (latest) version of the source code.
+
+replace (
```
Comment (jpkrohling) [inline go.work:205]: I assume this would cause the release process to change a bit. If that's the case, can we have an update to it as part of this PR?

Comment (mx-psi) [inline go.work:205]: Do we actually need these `replace` statements? My impression from reading https://go.dev/doc/tutorial/workspaces is that the `use` statements were enough

File: examples/tracing/Dockerfile (hunk 1 above)
Comment (jpkrohling) [reply examples/tracing/Dockerfile]: Answering my question: no.

File: go.work (hunk 2 above)
Comment (codeboten) [reply go.work]: Absolutely.

Comment (codeboten) [reply go.work]: These are no longer needed, they've been removed

**Comment by codeboten** [comment]
Closing this for now, there's no clear path to move forward with workspaces in this repo as it adds unnecessary complexity to the release process.

**Pull request closed without merging**